src/Apps/Settings/Routes/Purchases/Components/SettingsPurchasesRow.tsx

```diff
@@ -85,14 +96,61 @@ const getPaymentMethodText = (
   }
 }
 
+const getOrderActionButton = (
+  displayState: CommerceBuyerOrderOfferStateEnum,
+  orderId: string
+) => {
+  switch (displayState) {
+    case "PAYMENT_FAILED":
+      return (
+        <Button
+          // @ts-ignore
+          as={RouterLink}
+          to={`/orders/${orderId}/payment/new`}
+          variant="primaryBlack"
+          size="large"
+          width="50%"
+          minWidth={["240px", "200px"]}
+          my={[1, 0]}
+        >
+          Update Payment Method
+        </Button>
+      )
+    case "OFFER_RECEIVED":
+      return (
+        <Button
+          // @ts-ignore
+          as={RouterLink}
+          to={`/orders/${orderId}/status`}
+          variant="primaryBlack"
+          size="large"
+          width="50%"
+          minWidth={["240px", "200px"]}
+          my={[1, 0]}
+        >
+          Respond to Counteroffer
+        </Button>
+      )
+    default:
+      return null
+  }
+}
+
 interface SettingsPurchasesRowProps {
   order: SettingsPurchasesRow_order$data
 }
 
-const SettingsPurchasesRow: FC<React.PropsWithChildren<SettingsPurchasesRowProps>> = ({ order }) => {
+const SettingsPurchasesRow: FC<React.PropsWithChildren<
+  SettingsPurchasesRowProps
+>> = ({ order }) => {
   const [lineItem] = extractNodes(order?.lineItems)
   const { artwork, artworkVersion, fulfillments } = lineItem
-  const { requestedFulfillment } = order
+  const { requestedFulfillment, buyerAction } = order
+  let displayState: CommerceBuyerOrderOfferStateEnum = order.displayState
+
+  if (displayState == "SUBMITTED" && buyerAction == "OFFER_RECEIVED") {
+    displayState = "OFFER_RECEIVED"
+  }
```

Comment:
Just going back to my previous comment - I hope we can come to a permanent solution soon, maybe as part of checkout flow improvements. in the mean time i think i would name this variable something different from `displayState` since that is a specific value on the order from the server. `buyerState` seems right to me...